### PR TITLE
use the channel lib

### PR DESCRIPTION
**Pull request by GTrunSec**
response this issue to https://github.com/divnix/digga/pull/469

### CODE REVIEW

**Comment by GTrunSec** [comment]
We need a real channel to judge whether an overlay or isFunction.  https://github.com/gytis-ivaskevicius/flake-utils-plus/pull/120

File: lib/exportOverlays.nix [hunk 1]
```diff
@@ -60,7 +60,8 @@ let
       fakePkgs = {
         callPackage = it: it;
         isFakePkgs = true; # Overlay maintainers may throw an exception in case this key is set in case overlay is not compatible with this function.
-        lib = flake-utils-plus.lib.internal;
+        # import the lib input from first channel
+        lib =  if (builtins.attrNames channels) == [] then flake-utils-plus.lib.internal else inputs."${(builtins.elemAt (builtins.attrNames channels) 0)}".lib;
```
Comment (gytis-ivaskevicius) [inline lib/exportOverlays.nix:64]: How do we know what is first input? Also cant we simplify this as such:
```suggestion
        lib =  if (builtins.attrNames channels) == [] then flake-utils-plus.lib.internal else (builtins.elemAt (builtins.attrValues channels) 0).lib;
```

**Review by gytis-ivaskevicius — COMMENTED** [review]
May I have the exact scenario where existing implementation fails? Or review from @blaggacao  or @Pacman99?

**Comment by GTrunSec** [comment]
- https://github.com/gytis-ivaskevicius/flake-utils-plus/blob/master/lib/exportOverlays.nix#L80. 
- There is a problem with the way to judge an overlay function.

- tested by:  https://github.com/divnix/digga/pull/483/files#diff-15b26e32ee962cc4fbda5eea87245f72ed0b3710329e25e7a5c62a897610ce35R81

File: lib/exportOverlays.nix (hunk 1 above)
Comment (GTrunSec) [reply lib/exportOverlays.nix:64]: I don't know what you mean. IMO,  if we use an actual channel that should replace the flakePkg to https://github.com/divnix/digga/pull/483/files#diff-50b278a919472d6eab9c50638883df55c1129fb60f71ded49b06673428be8ceeR49

**Pull request closed without merging**